src/main/java/edu/njit/jerse/automation/RepositoryAutomationEngine.java

```diff
@@ -0,0 +1,239 @@
+            LOGGER.error("Error processing CSV file: {}", csvFilePath, e);
+            throw new RuntimeException("Error processing CSV file: " + csvFilePath, e);
+        }
+        LOGGER.info("Completed processing CSV file: {}", csvFilePath);
+    }
+
+    /**
+     * Reads a CSV file containing repository information and converts it into a list of
+     * {@code Repository} records. The CSV file must have a header specifying these columns.
+     * This method sets up the CSV format to skip the header record during parsing.
+     *
+     * @param csvFilePath the file path to the CSV file that contains the repository information
+     * @return a list of maps where each map contains the 'Repository' URL and the 'Branch' name
+     * @throws RuntimeException if an error occurs while trying to read or parse the CSV file. The original
+     *                          IOException is encapsulated within this RuntimeException.
+     */
+    private static List<Repository> readRepositoriesFromCsv(String csvFilePath) {
+        LOGGER.info("Reading CSV file: {}", csvFilePath);
+        List<Repository> repositories = new ArrayList<>();
+
+        CSVFormat csvFormat = CSVFormat.DEFAULT.builder().setHeader().setSkipHeaderRecord(true).build();
+
+        try (FileReader reader = new FileReader(csvFilePath);
+             CSVParser csvParser = new CSVParser(reader, csvFormat)) {
+
+            for (CSVRecord csvRecord : csvParser) {
+                Repository repoInfo = new Repository(csvRecord.get("Repository"), csvRecord.get("Branch"));
+                repositories.add(repoInfo);
+                LOGGER.info("Added repository: {}", repoInfo);
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error reading CSV file: {}", csvFilePath, e);
+            throw new RuntimeException("Error reading CSV file: " + csvFilePath, e);
+        }
+        return repositories;
+    }
+
+    /**
+     * Processes all repositories specified in the list. Each repository is cloned or fetched,
+     * and then further processed to apply automation routines.
+     *
+     * @param repositories a list of {@code Repository} records, each containing the 'Repository' URL and the
+     *                     'Branch' name
+     * @param repoDir      the directory where the repositories will be cloned or fetched
+     * @throws GitAPIException if a Git-specific error occurs during the cloning or fetching of the repositories
+     * @throws IOException     if an I/O error occurs during the access or manipulation of the repository files
+     */
+    private static void processAllRepositories(List<Repository> repositories, String repoDir)
+            throws GitAPIException, IOException {
+        LOGGER.info("Starting to process repositories");
+        for (Repository repository : repositories) {
+            String repoUrl = repository.url();
+            String branch = repository.branch();
+
+            if (repoUrl == null || branch == null) {
```

Comment:
Do this check either in the Repository constructor, or at Repository creation sites.  Then, there is never an invalid Repository created, and the error messages will more useful than ones at a use of a Repository.